src/pytest_allure_spec_coverage/spec_collectors/sphinx.py

```diff
@@ -0,0 +1,117 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Sphinx spec collector implementation"""
+import os
+import warnings
+from typing import Optional
+from docutils.core import publish_doctree
+from docutils.utils import SystemMessage
+
+from pytest_allure_spec_coverage.models.collector import Collector
+from pytest_allure_spec_coverage.models.scenario import Scenario, Parent
+
+
+class SphinxCollector(Collector):
+    """
+    Sphinx collector allows you to collect scenarios from .rst files
+    At the moment, only title will be read from .rst file
+    Template variables for title do not support
+
+    Possible options:
+        - sphinx_dir: required option where is places your scenarios
+        - sphinx_endpoint: where is hosted your scenarios. Need to form scenario link. Optionally
+    """
+
+    sphinx_dir: str
+    spec_endpoint: Optional[str]
+
+    def validate_config(self):
+        if "sphinx_dir" not in self.config:
+            raise ValueError("Option sphinx_dir is required")
+        sphinx_dir = self.config.get("sphinx_dir")
+        if not os.path.exists(sphinx_dir):
+            raise ValueError(f"Directory with sphinx specs {sphinx_dir} doesn't exists")
+
+        self.sphinx_dir = sphinx_dir
+        self.spec_endpoint = self.config.get("spec_endpoint")
+
+    def collect(self):
+        branch = os.getenv("BRANCH")
+        scenarios = []
+        parents_display_names = {}
+        root_path = self.sphinx_dir.rsplit("/", maxsplit=1)[0]
+        for root, _, files in os.walk(self.sphinx_dir):
+            parent_name = root.replace(root_path, "").strip("/").replace("/", ".")
+            parent_display_name = parent_name.rsplit(".", maxsplit=1)[-1]
+            if "index.rst" in files:
+                parent_display_name = self._get_title_from_rst(os.path.join(root, "index.rst")) or parent_display_name
+                files.remove("index.rst")
+            parents_display_names[parent_name] = parent_display_name
+
+            for file in filter(lambda f: f.endswith(".rst"), files):
+                name = file.rsplit(".", maxsplit=1)[0]
+                display_name = self._get_title_from_rst(os.path.join(root, file)) or file
+
+                scenarios.append(
+                    Scenario(
+                        name=name,
+                        display_name=display_name,
+                        parents=self._get_parents_by_fullname(parent_name, parents_display_names),
+                        link=self._create_link(name, parent_name, branch),
+                        branch=branch,
+                    )
+                )
+        return scenarios
+
+    def _create_link(self, name, parent_name, branch):
+        if not self.spec_endpoint:
+            return None
+        url = self.spec_endpoint
+        if branch:
+            url += f"/{branch}"
```

Comment:
Taking a branch into account is great! But in tests we use links without a branch. How we will match tests with specs in this case?
Also, if tests are executed on CI we will always get branch name

Reply:
We either need additional URL attribute like "display URL" or some method that will exclude branch from URL to be used in matching

Reply:
add `default_branch` option. This branch is unique on every project and will be configurable. If current branch is default, we don't put branch on spec url

Reply:
This is only one part of the issue. The second one is that we use links to the default branch only in test decorators. I guess we should invent a custom decorator and use only a part of the URL for matching and including in the new decorators.
But it seems to be a future issue.